composer.json

```diff
@@ -25,23 +25,6 @@
     "post-install-cmd": "if php -r 'exit( version_compare( phpversion(), \"8.1\", \">=\" ) ? 0 : 1 );'; then composer --working-dir=build-cs install --no-interaction; else echo 'Skipping composer install for build-cs since not on PHP 8.1+. You are running: '; php -v;  fi",
     "post-update-cmd": "if php -r 'exit( version_compare( phpversion(), \"8.1\", \">=\" ) ? 0 : 1 );'; then composer --working-dir=build-cs update --no-interaction; else echo 'Skipping composer update for build-cs since not on PHP 8.1+. You are running: '; php -v; fi",
     "phpstan": "build-cs/vendor/bin/phpstan analyse --memory-limit=2048M",
-    "format": "build-cs/vendor/bin/phpcbf --report-summary --report-source",
-    "lint": "build-cs/vendor/bin/phpcs",
-    "lint:all": [
-      "@lint",
-      "@lint:auto-sizes",
-      "@lint:dominant-color-images",
-      "@lint:embed-optimizer",
-      "@lint:optimization-detective",
-      "@lint:speculation-rules",
-      "@lint:webp-uploads"
-    ],
-    "lint:auto-sizes": "@lint -- ./plugins/auto-sizes --standard=./plugins/auto-sizes/phpcs.xml.dist",
-    "lint:dominant-color-images": "@lint -- ./plugins/dominant-color-images --standard=./plugins/dominant-color-images/phpcs.xml.dist",
-    "lint:embed-optimizer": "@lint -- ./plugins/embed-optimizer --standard=./plugins/embed-optimizer/phpcs.xml.dist",
-    "lint:optimization-detective": "@lint -- ./plugins/optimization-detective --standard=./plugins/optimization-detective/phpcs.xml.dist",
-    "lint:speculation-rules": "@lint -- ./plugins/speculation-rules --standard=./plugins/speculation-rules/phpcs.xml.dist",
-    "lint:webp-uploads": "@lint -- ./plugins/webp-uploads --standard=./plugins/webp-uploads/phpcs.xml.dist",
```

Comment:
Why are all of these being removed from here? Similar to @joemcgill's feedback in https://github.com/WordPress/performance/pull/1065#discussion_r1546419716, I think PHP based tooling makes sense to be accessible as a Composer script.

Can't this be solved in a PHP way? What's the benefit of moving all of these to NPM?

At a minimum, I think they should remain accessible via `composer.json` like before, while NPM could effectively be a "wrapper".

Reply:
@thelovekesh There would only be duplication if we have plugin specific scripts in both `composer.json` and `package.json`. What I'm proposing though is something like this:
* We abstract the PHPCS specific call in a custom PHP script that takes the plugin slug as a parameter and takes the potential `phpcs.xml` from the plugin folder into account.
* The `composer.json` can then use that instead of manually calling `phpcs` with the correct parameters.
* No changes to `package.json` would be needed.

PHPCS is PHP specific tooling, and `composer.json` is the appropriate location to manage such PHP specific tooling.

Reply:
@felixarntz Can you please take a look at the new changes? As suggested by @joemcgill, I have added a PHP script to handle linting in plugins and also added an NPM wrapper in composer commands.